jdbc/src/java/org/apache/hive/jdbc/HiveDatabaseMetaData.java

```diff
@@ -744,12 +744,16 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
     };
   }
 
-  public String getURL() throws SQLException {
-    throw new SQLFeatureNotSupportedException("Method not supported");
+  public String getURL() {
+    return connection.getConnectedUrl();
   }
 
   public String getUserName() throws SQLException {
-    throw new SQLFeatureNotSupportedException("Method not supported");
+    String userName = connection.getConnParams().getSessionVars().get(JdbcConnectionParams.AUTH_USER);
+    if ((userName == null) || userName.isEmpty()) {
+      userName = JdbcConnectionParams.ANONYMOUS_USER;
+    }
+    return userName;
   }
```

Comment:
can we just do
```
  public String getUserName() throws SQLException {
    return connection.getUserName();
  }
```

and make ``getUserName()`` in ``HiveConnection`` package-private (default)?

Reply:
Done.